### PR TITLE
Stream container statistics

### PR DESCRIPTION
**Pull request by wendigo**
This PR changes the way statistics are fetched from Docker: instead of closing the stream every time statistics are fetched, the stream remains open (docker sends them every 1s) until the environment is stopped. Arriving values are used to calculate deltas.

This PR also improves the formatting of statistics values so they are displayed in a table:

Screenshot 2020-11-17 at 20 27 46

### CODE REVIEW

**Review by kokosing — COMMENTED** [review]
lgtm

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java:269]:
```
.mapToInt(String::length)
.max()
.orElse(0);
```

what about

**Comment by findepi** [comment]
Couldn't we simply remove the use of AsyncTimeout in `io.prestosql.tests.product.launcher.env.StatisticsFetcher#get` by moving everything into the `executor`
https://github.com/prestosql/presto/blob/d0c206c9566cce04766b4658ba9b6be93cdf66f5/presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java#L55-L59
?

**Comment by wendigo** [comment]
@findepi tried that, doesn't work either. This approach seems to work - it opens a single stream for every container which is nicely closed when the environment is stopped.

**Comment by findepi** [comment]
let's get  https://github.com/prestosql/presto/pull/5989 in and let's stress test this one more before merging

**Comment by wendigo** [comment]
Fixes https://github.com/prestosql/presto/issues/5422

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java:236]: nit: closing and then printing seems more natural.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:45]: final

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:46]: final

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java:236]: makes sense

**Review by kokosing — APPROVED** [review]

**Comment by wendigo** [comment]
Please do not merge it yet - I'll push updated version (i/o needs to be converted to succinct form):

Screenshot 2020-11-18 at 11 48 19

**Comment by wendigo** [comment]
Screenshot 2020-11-18 at 12 07 51

**Comment by wendigo** [comment]
I haven't seen any hang with this PR (5 or 6 consecutive builds).

**Review by losipiuk — COMMENTED** [review]
I found some pending comments from yesterday :)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:75]: It’s not but I can add an assertion here

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:137]: It will log always (when container is stopped we will get recv() eof error)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java]: It is now

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:150]: We need if we close before stream ends (ie when pt timeout occurs)

**Comment by wendigo** [comment]
AC @losipiuk

**Review by losipiuk — APPROVED** [review]

**Pull request merged**
